### PR TITLE
fix(custom-measurements): Update custom measurement logic

### PR DESCRIPTION
**Pull request by wmak**
- This updates the lgoic that determines if a mri represents a custom measurements to instead of checking for a custom namespace, that its a transaction MRI, but one that isn't in the hardcoded list in TransactionMRI

### CODE REVIEW

File: src/sentry/snuba/metrics/naming_layer/mri.py [hunk 1]
```diff
@@ -124,4 +128,15 @@ def parse_mri(mri_string: str) -> Optional[ParsedMRI]:
...
+
+    This checks the namespace, and name to match what we expect first before iterating through the
+    members of the transaction MRI enum to make sure it isn't a standard measurement
+    """
+    return (
+        parsed_mri.namespace == "transactions"
+        and parsed_mri.name.startswith("measurements.")
+        and
+        # Iterate through the transaction MRI and check that this parsed_mri isn't in there
+        parsed_mri.mri_string not in [mri.value for mri in TransactionMRI.__members__.values()]
```
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply src/sentry/snuba/metrics/naming_layer/mri.py:141]: so you can use here:

```
str(parsed_mri)
```

cause `parsed_mri.mri_string` looks too wordy to me.

File: src/sentry/snuba/metrics/naming_layer/mri.py [hunk 2]
```diff
@@ -113,6 +113,10 @@ class ParsedMRI:
     name: str
     unit: str
 
+    @property
+    def mri_string(self) -> str:
+        return f"{self.entity}:{self.namespace}/{self.name}@{self.unit}"
```
Comment (andriisoldatenko) [inline src/sentry/snuba/metrics/naming_layer/mri.py:118]: also i don't see unit-test for this line.

File: src/sentry/snuba/metrics/naming_layer/mri.py [hunk 3]
```diff
@@ -113,6 +113,10 @@ class ParsedMRI:
     name: str
     unit: str
 
+    @property
+    def mri_string(self) -> str:
```
[the remark this answers is not in the document]
Comment (wmak) [reply src/sentry/snuba/metrics/naming_layer/mri.py:117]: 🤔 feels weird to me that the `str` of a *parsed* MRI is just the MRI again though? But Maybe I'm alone in that

File: src/sentry/snuba/metrics/naming_layer/mri.py (hunk 2 above)
Comment (wmak) [reply src/sentry/snuba/metrics/naming_layer/mri.py:118]: 🤷 doesn't feel necessary to me since its a string format & already implicitly tested by is_custom_measurement. But I added an assert to the parse_mri test anyways

Comment (andriisoldatenko) [reply src/sentry/snuba/metrics/naming_layer/mri.py:118]: agreed.

File: src/sentry/snuba/metrics/naming_layer/mri.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (andriisoldatenko) [reply src/sentry/snuba/metrics/naming_layer/mri.py:117]: it was just suggestion, i;m fine with custom name. Only is it possible to remove `_mri`? Cause dataclass already has mri in his name.

**Review by edwardgou-sentry — APPROVED** [review]
works for me

**Review by ahmedetefy — APPROVED** [review]
lgtm! thanks for the fix!

File: src/sentry/snuba/metrics/naming_layer/mri.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ahmedetefy) [reply src/sentry/snuba/metrics/naming_layer/mri.py:141]: @wmak yeah I know what you mean, I have run into this before 

I think I will put up a follow up PR with something like this https://stackoverflow.com/a/65225753

**Review by andriisoldatenko — APPROVED** [review]
